scout/server/blueprints/cases/templates/cases/case.html

```diff
@@ -229,6 +230,67 @@ <h4 class="card-header">Case: {{case.display_name}}</h4>
   </div>
 {% endmacro %}
 
+{% macro group_panel() %}
+  <div class="card panel-default">
+    <div data-toggle='tooltip' class="panel-heading" title=""><span class="fa fa-users"></span>&nbsp;Group connected cases</div>
+    <div class="card-body">
+      {% if case_groups %}
+        {% for group_id, case_group in case_groups.items() %}
+          {{ modal_add_to_group(group_id) }}
+          <div class="row">
+            <div class="col">
+              <span class="fa fa-users"></span>&nbsp;
+            </div>
+            <div class="col text-right">
+              <div class="btn-group">
+                <button type="button" class="btn btn-outline-secondary form-control" data-toggle="modal" data-target="#add-case-group-{{group_id}}">
+                  <span class="fa fa-plus-square-o"></span>&nbsp;Add to group
+                </button>
+              </div>
+            </div>
+          </div>
+          <ul class="list-group">
+          {% for grouped_case in case_group %}
+            <li class="list-group-item">
+                <a href="{{ url_for('cases.case', institute_id=grouped_case.owner, case_name=grouped_case.display_name) }}"">{{ grouped_case.display_name }}</a>
+                <span class="pull-right">
+                  <a href="{{ url_for('cases.remove_case_group', institute_id=institute._id, case_name=grouped_case.display_name, case_group=group_id) }}" class="btn btn-link btn-sm">
+                    <span class="fa fa-remove"></span></a>
+                </span>
+            </li>
+          {% endfor %}
+          </ul>
+        {% endfor %}
+      {% endif %}
+      <a href="{{ url_for('cases.add_case_group',institute_id=institute._id, case_name=case.display_name) }}" class="btn btn-primary btn-xs text-white"><span class="fa fa-user-plus text-white mr-3"></span>Add group</a>
```

Comment:
> Why there are 2 distinct buttons to add a case to a group? If you click several times on the one on the bottom you find yourself in this situation, which is a bit weird

Thanks a bunch for the review! Super to see what is not obvious to others.

The underlying reason is that a case should be able to join multiple groups, to allow for different investigations, like one with a related family, one with cancer remission sample, one with another tumor or sth. Also, groups don't get made on load (yet?) so they have to be made through the interface at some point.

There are three kinds of buttons, and in my simple mind they form a nice minimum, though perhaps we can come up with better names for them:
- "Add group" adds a new group (unnamed, can add names to see if that helps make it more clear) - and it is assumed the current case is always going to be in that new group, so it is automatically bound into the group. Groups are after all only shown on the case pages if the case belongs to the group. Perhaps it could be named "Create group"?
- "Add to group" adds other cases to the selected group. We agree on this one. 😸 Perhaps it could be named "Add other case" or such?
- "X" on group list item cases removes the case from the pertinent group. If no cases are left in the group, the group is deleted. If there are still other cases in the group, but present case A is removed, it just leaves the group, and the group is no longer visible on the case page of A. It will still be visible from the other group pages.

Reply:
I did understand that one case might belong to different groups, but to me it is a bit confusing that if you want to assign a case (case x) to a group with other 2 cases then you should do it from the case page of the other 2 cases, because if you do it from the page of case x it's only assigned to one case, in a new group, and not where you want.. But probably this will be more clear if one assigns a name to the groups..

Reply:
Well, you can do it from that third case page, but you then need to mention both the two other previously added cases that you wanted in that group. More action economy would be gained by adding the single case from either of the two already in a group. Fair enough to me, and nothing that can't be fixed, but let's see what happens with just a rename.

Reply:
Ok, see what you think - I just added a placeholder name based off the ObjectId, much like git commits. We *can* make an editable name, but it will add some complexity. I feel just having some more separator beyond an icon and a button between the different groups might have done the trick for now. If you still think the name request is inevitable, I'll add it anyway.

![Screenshot 2021-01-20 at 16 17 49](https://user-images.githubusercontent.com/758570/105195730-34163200-5b3b-11eb-9743-0393e3efeb63.png)

Reply:
Mmm I don't think displaying the _id of the groups is much of help here. I think it would be best not to have it if it can't be chosen by the user, sorry. I guess we can go with this system and fix it later if they want to specify a custom name for the group?

I thinking for instance at different types of analyses for the same case, or different sequenced tissues for a cancer case and so on. It would be useful to have a way to name these groups, but perhaps I'm overthinking and it's enough to be able to assign them together!